main/lsp/lsp_helpers.cc

```diff
@@ -44,6 +44,11 @@ bool hideSymbol(const core::GlobalState &gs, core::SymbolRef sym) {
     if (name.kind() == core::NameKind::UNIQUE && name.dataUnique(gs)->original == core::Names::blockTemp()) {
         return true;
     }
+
+    if (name == core::Names::requiredAncestorsLin()) {
+        return true;
+    }
```

Comment:
`hideSymbol` is used in few more places then the problematic [codepath](https://github.com/sorbet/sorbet/blob/5b4782ef14d493de57db52a314c5e1fb1a3018ec/main/lsp/requests/document_symbol.cc#L217) but I think that's okay.

Reply:
![Screenshot 2024-08-01 at 11 47 24 AM](https://github.com/user-attachments/assets/5eba4e40-a9a7-4dad-8052-75140fda3285)

Offline chat: This is a synthetic method but it does take the user to where `requires_ancestor` is called so it might be a useful feature.